### PR TITLE
Added asNumber to OrderByColumnSpec

### PR DESCRIPTION
**Pull request by flowbehappy**
This commit make dimension can be sorted numerically(only "long" here) besides lexicographically. e.g
```
query:
...
"dimensions": [
	"id",
],
"limitSpec": {
	"type": "default",
	"limit": 20,
	"columns": [
		{
			"asNumber": true,
			"dimension": "id",
			"direction": "ASCENDING"
		}
	]
},
...

result:
id	name	score
0	alias	90
1	bob		75
2	joe		60
11	flow	88
14	kevin	54
```

### CODE REVIEW

**Comment by flowbehappy** [comment]
@drcrallen @nishantmonu51 I added test case and did some code clean up. And sorry for taking so long.

**Comment by drcrallen** [comment]
Thanks @flowbehappy  few more comments:

If a dimension is called "id" I would expect it to be a high cardinality dimension rather than just "`default`". Is "id" supposed to be relatively high cardinality in this case?

**Comment by flowbehappy** [comment]
@drcrallen Totally agree.  I will generate another more real test data.

**Comment by fjy** [comment]
@cheddar @himanshug @xvrl any thoughts here? My biggest high level concern is that it only Longs are supported, but it is a useful feature.

File: processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java [hunk 1]
```diff
@@ -149,35 +154,100 @@ public int compare(Row left, Row right)
...
+                  if (dimList.isEmpty()) {
+                    return null;
+                  }
+
+                  String dimValue = dimList.get(0);
+                  // Make sure dimValue is number parsable.
+                  if (dimValue.isEmpty()) {
+                    return dimValue;
+                  }
+                  for (int i = 0; i < dimValue.length(); i++) {
```
Comment (himanshug) [inline processing/src/main/java/io/druid/query/groupby/orderby/DefaultLimitSpec.java]: It does not seem to hard to make this feature complete by adding support for doubles as well.
We can use a regex here instead to detect if its a double value or long value or something else. Accordingly use Double.parseDouble(..) or Long.parseLong(..) .
And in the compare above add another case to handle Doubles.
Also, you can refactor this code a little bit by having a private method like
private Comparable getDimensionValue() {

}
and calling return getDimensionValue(dimList.get(0));

**Comment by gianm** [comment]
I believe this use case is addressed by #3270 through "dimensionOrder": "numeric".

**Pull request closed without merging**